pubtools/_content_gateway/cgw_client.py

```diff
@@ -0,0 +1,464 @@
+import json
+from .cgw_authentication import CGWBasicAuth
+from .cgw_session import CGWSession
+
+class CGWClientError(Exception):
+    """
+    Custom exception to handle Content Gateway Client errors
+    """
+
+    def __init__(self, message):
+        self.message = message
+
+
+class CGWClient:
+    def __init__(self, hostname, cgw_auth=None, verify=True):
+        self.hostname = hostname.rstrip("/")
+        self.cgw_session = CGWSession(self.hostname)
+        if cgw_auth:
+            cgw_auth.make_auth(self.cgw_session)
+
+    def call_cgw_api(self, method, endpoint, params=None, data=None):
+        # Check if CGW hostname is present
+        if not self.hostname:
+            raise CGWClientError("No content gateway hostname found")
+
+        # Check if correct method id passed
+        if method not in ["GET", "POST", "PATCH", "PUT", "DELETE"]:
+            raise CGWClientError("Wrong request method passed")
+
+        response = None
+        output = True
```

Comment:
I don't understand why you set it to True. Method should return one type of result. In this case python dict. In the case there is nothing to return (like for DELETE), return empty dict.